router/functionServiceMap.go

```diff
@@ -75,3 +75,17 @@ func (fmap *functionServiceMap) assign(f *api.ObjectMeta, serviceUrl *url.URL) {
 		// ignore error
 	}
 }
+
```

Comment:
Why do we need to invalidate the function service map? If the function's changed, it's resource version has changed. And the function service map key contains the resource version, which means a lookup of an existing function with a new resource version will actually be a cache miss.